tests/test_utils.go

```diff
@@ -0,0 +1,157 @@
+package tests
+
+import (
+	"bytes"
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+)
+
+type TestUtilsImpl struct {
+}
+
+func (t TestUtilsImpl) setup_expectation(requestToMock *string, responseToGet *string) *http.Response {
+
+	type Times struct {
+		remainingTimes int
+		unlimited      bool
+	}
+
+	type Body struct {
+		httpRequest  string
+		httpResponse string
+		times        Times
+	}
+
+	url := "http://localhost:1080/expectation"
+
+	body := Body{
+		httpRequest:  *requestToMock,
+		httpResponse: *responseToGet,
+		times: Times{
+			remainingTimes: 1,
+			unlimited:      false,
+		},
+	}
+
+	client := &http.Client{}
+
+	json, err := json.Marshal(body)
+	if err != nil {
+		panic(err)
+	}
+
+	req, err := http.NewRequest(http.MethodPut, url, bytes.NewBuffer(json))
+	if err != nil {
+		panic(err)
+	}
+
+	req.Header.Set("Content-Type", "application/json; charset=utf-8")
+
+	resp, err := client.Do(req)
+	if err != nil {
+		panic(err)
+	}
+
+	return resp
+
+}
+
+func (t TestUtilsImpl) verify_expectation_matched_times(expectationId string, times int) *http.Response {
+
+	url := "http://localhost:1080/verify"
+
+	type ExpectationId struct {
+		id string
+	}
+
+	type Times struct {
+		atLeast int
+		atMost  int
+	}
+
+	type Body struct {
+		expectationId ExpectationId
+		times         Times
+	}
+
+	body := Body{
+		expectationId: ExpectationId{
+			id: "",
+		},
+		times: Times{
+			atLeast: times,
+			atMost:  times,
+		},
+	}
+
+	client := &http.Client{}
+
+	json, err := json.Marshal(body)
+	if err != nil {
+		panic(err)
+	}
+
+	req, err := http.NewRequest(http.MethodPut, url, bytes.NewBuffer(json))
+	if err != nil {
+		panic(err)
+	}
+
+	req.Header.Set("Content-Type", "application/json; charset=utf-8")
+
+	resp, err := client.Do(req)
+	if err != nil {
+		panic(err)
+	}
+
+	return resp
+}
+
+func (t TestUtilsImpl) reset_expectations() {
+	url := "http://localhost:1080/reset"
+
+	http.NewRequest(http.MethodPut, url, http.NoBody)
+
+}
+
+func (t TestUtilsImpl) generate_payloads_from(filename string, payloadsPath string) (request Request, response Response) {
+
+	if payloadsPath == "" {
+		payloadsPath = "./payloads"
+	}
+
+	var payload Payload
+
+	file, err := ioutil.ReadFile("#{payloadsPath}/#{filename}.json")
+	if err != nil {
+		panic(err)
+	}
+
+	json.Unmarshal(file, &payload)
+
+	return payload.request, payload.response
+
+}
+
+func (t TestUtilsImpl) generate_query_params(request Request) Opts {
```

Comment:
Verifying, but how would this method's return value be used? Would the values inside `opts.Opt` be retrieved one by one? Also, would it be better to return a `map[string][string]` directly instead?

Reply:
im verfying the possible use of using Context here instead, if not ill just simply return `map[string][string]`

Reply:
i think this one can be resolved